python_modules/libraries/dagster-components/dagster_components_tests/unit_tests/test_registry.py

```diff
@@ -0,0 +1,114 @@
+import os
+import subprocess
+from pathlib import Path
+
+from dagster_components import ComponentRegistry
+
+
+def test_components_from_dagster():
+    registry = ComponentRegistry.from_entry_point_discovery()
+    assert registry.has("dagster_components.dbt_project")
+    assert registry.has("dagster_components.sling_replication")
+    assert registry.has("dagster_components.pipes_subprocess_script_collection")
+
+
+def _find_repo_root():
+    current = Path(__file__).parent
+    while not (current / ".git").exists():
+        if current == Path("/"):
+            raise Exception("Could not find the repository root.")
+        current = current.parent
+    return current
+
+
+repo_root = _find_repo_root()
+
+# Our pyproject.toml installs local dagster components
+PYPROJECT_TOML = f"""
+[build-system]
+requires = ["setuptools", "wheel"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "dagster-foo"
+version = "0.1.0"
+description = "A simple example package"
+authors = [
+    {{ name = "Your Name", email = "your.email@example.com" }}
+]
+dependencies = [
+    "dagster",
+    "dagster-components",
+    "dagster-dbt",
+    "dagster-embedded-elt",
+]
+
+[tool.uv.sources]
+dagster = {{ path = "{repo_root}/python_modules/dagster" }}
+dagster-pipes = {{ path = "{repo_root}/python_modules/dagster-pipes" }}
+dagster-components = {{ path = "{repo_root}/python_modules/libraries/dagster-components" }}
+dagster-dbt = {{ path = "{repo_root}/python_modules/libraries/dagster-dbt" }}
+dagster-embedded-elt = {{ path = "{repo_root}/python_modules/libraries/dagster-embedded-elt" }}
```

Comment:
Unfortunately need to include all this here currently because the dbt/sling components are in the `dagster-components` lib folder and get loaded when it's crawled.